lib/bolt/shell/bash.rb

```diff
@@ -390,14 +389,20 @@ def execute(command, sudoable: false, **options)
           # See if we can read from out or err, or write to in
           ready_read, ready_write, = select(read_streams.keys, write_stream, nil, timeout)
 
-          # Read from out and err
           ready_read&.each do |stream|
+            stream_name = stream == out ? 'out' : 'err'
             # Check for sudo prompt
-            read_streams[stream] << if use_sudo
-                                      check_sudo(stream, inp, options[:stdin])
-                                    else
-                                      stream.readpartial(CHUNK_SIZE)
-                                    end
+            to_print = if use_sudo
+                         check_sudo(stream, inp, options[:stdin])
+                       else
+                         stream.readpartial(CHUNK_SIZE)
+                       end
+
+            if !to_print.chomp.empty? && @stream_logger
+              @stream_logger.warn("[#{@target.safe_name}] #{stream_name}: #{to_print.chomp}")
```

Comment:
When the output has newlines the target & stream names get lost pretty easily. I wonder if adding these to the beginning of each line, or indenting each line would help with readability and quickly figuring out where output is coming from?

Reply:
Oo definitely. Great idea.